walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,78 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
```

Comment:
A general implementation suggestion:
Your current impl. generates two lists of occupied slots (for mandatory and all attendees), which you then have to send twice to the getOptionalSlots method and do the math yourself.
I propose changing the implementation to use https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Range.html (see if the import works, not sure about that).

Using Range will make your life a whole lot easier. You can start by having a Duration range of an entire day, and use methods like intersection to do the job for you.

Again, this is a suggestion, not a must.

Reply:
A little bit of a delay treating this comment because I tried to change the implemenatation to something like you suggested. The Range import didn't work but I still tried to do something more efficeint. I tried different things but eventually came up with only a slight change that makes it more efficient, but not something major. What I changed makes the sorting more efficient- it sorts the mandatory slots and the optinal seperatly, and then merges them to a sorted list of all slots. This is instead of what I was doing before- which is sort only madatory  and sort all.